src/libraries/System.Net.Requests/src/System/Net/HttpWebResponse.cs

```diff
@@ -337,7 +339,35 @@ public override Stream GetResponseStream()
             CheckDisposed();
             if (_httpResponseMessage.Content != null)
             {
-                return _httpResponseMessage.Content.ReadAsStream();
+                Stream contentStream = _httpResponseMessage.Content.ReadAsStream();
+                if (_maxErrorResponseLength < 0 || StatusCode < HttpStatusCode.BadRequest)
+                {
+                    return contentStream;
+                }
+
+                MemoryStream memoryStream = new MemoryStream();
+                byte[] buffer = new byte[1024];
+                int readLength = 0;
+
+                while (readLength < _maxErrorResponseLength)
+                {
+                    int len = contentStream.Read(buffer, 0, Math.Min(_maxErrorResponseLength - readLength, buffer.Length));
+                    if (len == 0)
+                    {
+                        break;
+                    }
+                    memoryStream.Write(buffer, 0, len);
+                    readLength += len;
+                }
+                memoryStream.Seek(0, SeekOrigin.Begin);
+                try
+                {
+                    return memoryStream;
+                }
+                finally
```

Comment:
The `return` can't throw here - the reads above can

Reply:
my intention was to call `Dispose` on `contentStream` after returning